int16[] data
int32[] time_offset